frontend/packages/console-shared/src/types/route-params.ts

```diff
@@ -0,0 +1,14 @@
+export type ConsoleRouteParams = Partial<
```

Comment:
Would it be better to have specific types that for params expected on specific pages? (For example, a resource details page wouldn't ever have a param like `currentCSV`.)

Reply:
Yes, we can prefer composition of context specific types instead of aggregating everything in a context agnostic type.

I'll update the code.